timemachine/optimize/protocol.py

```diff
@@ -115,7 +116,7 @@ def log_weights_from_mixture(u_kn: Array, f_k: Array, N_k: Array) -> Array:
     mixture of states p(x) = (1 / K) \sum_k e^-u_k / Z_k
     """
     log_q_k = f_k - u_kn.T
-    N_k = jnp.array(N_k, dtype=jnp.float64)  # may be ints, or in a list...
+    N_k = np.array(N_k, dtype=np.float64)  # may be ints, or in a list...
```

Comment:
actually why are these `np.float64` dtypes?

Reply:
not a strong reason, but didn't modify this instance since passing to `jax.scipy.logsumexp(..., b=N_k)` in the next line

there's not an error if these are ints rather than floats (I think in earlier version there was), and it doesn't matter if these are np arrays or jnp arrays, but it may be marginally clearer / more general to emphasize that the mixture weights can be arbitrary floats (that in this case just happen to be proportional to ints)

Reply:
thanks! forgot that weights in `logsumexp` must be floats!